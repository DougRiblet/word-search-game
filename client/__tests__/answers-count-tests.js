import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import AnswersCount from '../components/Answers-Count/answers-count-present';

describe('Answers Count', () => {
  test('Snapshot', () => {
    const AnswersCount_tree = renderer.create(
      <AnswersCount
      poollength={14}
      foundlength={8}
      allownew={false}
      endGame={() => endGame()}
    />
    ).toJSON();
    expect(AnswersCount_tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AnswersCount
      poollength={14}
      foundlength={8}
      allownew={false}
      endGame={() => endGame()}
    />, div);
  });

  test('displays two counts correctly', () => {
    const wrapper = shallow(<AnswersCount poollength={14} foundlength={8} />);
    expect(wrapper.text()).toEqual('8 of 14');
    expect(wrapper.html()).toEqual('<div id="answers-count">8 of 14</div>');
  });

  test('calls endGame when all words found in active game', () => {
    const endGame = jest.fn();
    const wrapper = mount(
      <AnswersCount
        poollength={14}
        foundlength={14}
        allownew={false}
        endGame={endGame}
      />
    );
    expect(endGame).toHaveBeenCalled();
  });

});
