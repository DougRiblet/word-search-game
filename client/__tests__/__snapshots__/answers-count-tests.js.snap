// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Count Snapshot 1`] = `
<div
  id="answers-count"
>
  8
   of 
  14
</div>
`;
